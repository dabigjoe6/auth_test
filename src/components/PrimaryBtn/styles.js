import {colors} from '../../config';

const styles = {
  container: {
    width: '100%',
    height: 50,
    backgroundColor: colors.primary,
    alignItems: 'center',
    justifyContent: 'center',
    marginVertical: 10,
    borderRadius: 4,
  },
  transparentContainer: {
    width: '100%',
    height: 50,
    backgroundColor: 'transparent',
    alignItems: 'center',
    justifyContent: 'center',
    marginVertical: 10,
  },
  title: {
    color: 'white',
    fontSize: 16,
  },
  transparentTitle: {
    color: colors.primary,
  },
};

export default styles;
