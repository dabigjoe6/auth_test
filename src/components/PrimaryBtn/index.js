import React from 'react';
import {Pressable, Text, View} from 'react-native';
import styles from './styles';

const PrimaryBtn = ({onPress, isLoading, title, transparent}) => {
  const handlePress = () => {
    if (!isLoading) {
      onPress();
    }
  };
  return (
    <Pressable onPress={handlePress} testID="PrimaryBtn">
      <View
        style={transparent ? styles.transparentContainer : styles.container}>
        <Text style={transparent ? styles.transparentTitle : styles.title}>
          {isLoading ? 'Loading...' : title}
        </Text>
      </View>
    </Pressable>
  );
};

export default PrimaryBtn;
