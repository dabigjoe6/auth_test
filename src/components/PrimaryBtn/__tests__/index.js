import React from 'react';

import renderer from 'react-test-renderer';
import PrimaryBtn from '../index';
import {View, Text} from 'react-native';
import {shallow} from 'enzyme';

describe('Primary Button', () => {
  it('should render without issues', () => {
    const component = renderer.create(<PrimaryBtn />);
    const instance = component.root;

    expect(
      instance.findAll((comp) => comp.props.testID === 'PrimaryBtn').length,
    ).toBeGreaterThanOrEqual(1);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should be call onPress event', () => {
    const result = 'I was Pressed';
    const mockFn = jest.fn(() => result);
    const component = renderer.create(<PrimaryBtn onPress={mockFn} />);

    expect(component.root.props.onPress).not.toHaveBeenCalled();

    component.root.props.onPress();

    expect(component.root.props.onPress.mock.calls.length).toBe(1);
    expect(component.root.props.onPress()).toBe(result);
  });

  it('should not call onPress event when loading is true', () => {
    const result = 'I was pressed';
    const mockFn = jest.fn(() => result);

    const wrapper = shallow(<PrimaryBtn isLoading={false} onPress={mockFn} />);

    wrapper.setProps({isLoading: true});
    expect(wrapper.props('isLoading')).toBeTruthy();
    wrapper.simulate('press');
    expect(mockFn.mock.calls.length).toBe(0);
  });

  it('should be Loading... text when isLoading === true', () => {
    const result = 'I was pressed';

    const mockFn = jest.fn(() => result);

    const wrapper = shallow(<PrimaryBtn isLoading={false} onPress={mockFn} />);
    wrapper.setProps({isLoading: true});
    wrapper.update();
    expect(wrapper.props('isLoading')).toBeTruthy();
    expect(wrapper.contains('Loading...')).toBeTruthy();
  });

  it('should have transparent styles when transparent is true', () => {
    const transparentContainer = {
      width: '100%',
      height: 50,
      backgroundColor: 'transparent',
      alignItems: 'center',
      justifyContent: 'center',
      marginVertical: 10,
    };

    const transparentTitle = {
      color: '#74B3CE',
    };

    const wrapper = shallow(<PrimaryBtn transparent />);
    expect(wrapper.children(View).props('style').style).toMatchObject(
      transparentContainer,
    );

    expect(
      wrapper.children(View).children(Text).props('style').style,
    ).toMatchObject(transparentTitle);
  });
});
