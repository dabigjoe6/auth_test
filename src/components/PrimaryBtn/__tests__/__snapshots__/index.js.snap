// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Button should render without issues 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="PrimaryBtn"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#74B3CE",
        "borderRadius": 4,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
        }
      }
    />
  </View>
</View>
`;
